src/algorithm/simplify.rs

```diff
@@ -0,0 +1,111 @@
+use num::Float;
+use types::{Point, LineString};
+use algorithm::distance::Distance;
+
+// perpendicular distance from a point to a line
+fn point_line_distance<T>(point: &Point<T>, start: &Point<T>, end: &Point<T>) -> T
+    where T: Float 
+{
+    if start == end {
+        return point.distance(&start)
+    } else {
+        let numerator = ((end.x() - start.x()) * (start.y() - point.y()) -
+                 (start.x() - point.x()) * (end.y() - start.y()))
+            .abs();
+            let denominator = start.distance(&end);
+        numerator / denominator
+    }
+}
+
+// Ramer–Douglas-Peucker line simplification algorithm
+fn rdp<T>(points: &[Point<T>], epsilon: &T) -> Vec<Point<T>> 
+    where T: Float
+{
+    let mut dmax = T::zero();
+    let mut index: usize = 0;
+    let mut distance: T;
+
+    for (i, _) in points.iter().enumerate().take(points.len() - 1).skip(1) {
+        distance = point_line_distance(&points[i],
+                                       &*points.first().unwrap(),
```

Comment:
Instead of `.first()`? Yep, I just instinctively shy away from array indexing, but we know it'll be OK in this case.